src/ahbicht/expressions/condition_expression_parser.py

```diff
@@ -53,3 +57,48 @@ def parse_condition_expression_to_tree(condition_expression: str) -> Tree:
         ) from eof
 
     return parsed_tree
+
+
+def find_prerequisites(tree_or_list: Union[Tree, List[str]], sanitize: bool = False) -> ContentEvaluationPrerequisites:
+    """
+    find different types of condition nodes inside the given tree or list of keys.
+    The types are differentiated by their number range.
+    See 'Allgemeine Festlegungen' from EDI@Energy.
+    """
+    result = ContentEvaluationPrerequisites(
+        format_constraint_keys=[], requirement_constraint_keys=[], hint_keys=[], package_keys=[]
+    )
+    condition_keys: List[str]
+    if isinstance(tree_or_list, list):
+        condition_keys = tree_or_list
+    else:  # it's tree, hopefully
+        condition_keys = [
+            x.value  # type:ignore[attr-defined]
+            for x in tree_or_list.scan_values(
+                lambda token: token.type == "INT"  # type:ignore[union-attr]
+            )
+        ]
+    for condition_key in condition_keys:
+        condition_node_type = derive_condition_node_type(condition_key)
+        if condition_node_type == ConditionNodeType.REQUIREMENT_CONSTRAINT:
+            result.requirement_constraint_keys.append(condition_key)
+        elif condition_node_type == ConditionNodeType.HINT:
+            result.hint_keys.append(condition_key)
+        elif condition_node_type == ConditionNodeType.FORMAT_CONSTRAINT:
+            result.format_constraint_keys.append(condition_key)
+        elif condition_node_type == ConditionNodeType.PACKAGE:
+            result.package_keys.append(condition_key)
```

Comment:
I would suggest to use `is` instead of `==`. We are already using ENUMs so we can perform a strict "comparison"
```suggestion
        if condition_node_type is ConditionNodeType.REQUIREMENT_CONSTRAINT:
            result.requirement_constraint_keys.append(condition_key)
        elif condition_node_type is ConditionNodeType.HINT:
            result.hint_keys.append(condition_key)
        elif condition_node_type is ConditionNodeType.FORMAT_CONSTRAINT:
            result.format_constraint_keys.append(condition_key)
        elif condition_node_type is ConditionNodeType.PACKAGE:
            result.package_keys.append(condition_key)
```

Reply:
works as designed 👍